src/search/search-index-new/storage/types.ts

```diff
@@ -39,6 +50,18 @@ export interface ManageableStorage extends RegisterableStorage {
     initialized: boolean
     registry: StorageRegistry
     putObject(collectionName: string, object): Promise<void>
+    findObject<T>(collectionName: string, filter: FilterQuery<T>): Promise<T>
+    findAll<T>(collectionName: string, filter: FilterQuery<T>): Promise<T[]>
```

Comment:
@poltak I think here we have to make third arguments (`findOpts`) optional, otherwise we will not able to send the third argument, or we cannot change the `findOpts` as the globally.